feature/preview/src/main/java/com/google/jetpackcamera/feature/preview/ui/PreviewScreenComponents.kt

```diff
@@ -0,0 +1,223 @@
+                            Log.d(TAG, "onTap $offset")
+                        } catch (e: UninitializedPropertyAccessException) {
+                            Log.d(TAG, "onTap $offset")
+                            e.printStackTrace()
+                        }
+                    }
+                )
+            },
+
+        contentAlignment = Alignment.Center
+    ) {
+        val maxAspectRatio: Float = maxWidth / maxHeight
+        val aspectRatio: Float =
+            previewUiState.currentCameraSettings.aspect_ratio.ratio.toFloat()
+        val shouldUseMaxWidth = maxAspectRatio <= aspectRatio
+        val width = if (shouldUseMaxWidth) maxWidth else maxHeight * aspectRatio
+        val height = if (!shouldUseMaxWidth) maxHeight else maxWidth / aspectRatio
+        Box(
+            modifier = Modifier
+                .width(width)
+                .height(height)
+                .transformable(state = transformableState)
+
+        ) {
+            CameraPreview(
+                modifier = Modifier
+                    .fillMaxSize(),
+                onSurfaceProviderReady = onSurfaceProviderReady,
+                onRequestBitmapReady = {
+                    val bitmap = it.invoke()
+                },
+                setSurfaceView = { s: View ->
+                    viewInfo = s
+                }
+            )
+        }
+    }
+}
+
+@Composable
+fun FlipCameraButton(
+    modifier: Modifier = Modifier,
+    enabledCondition: Boolean,
+    onClick: () -> Unit
+) {
+    Box(modifier = modifier) {
+        IconButton(
+            modifier = Modifier
+                .align(Alignment.Center)
+                .size(40.dp),
+            onClick = onClick,
+            enabled = enabledCondition
+        ) {
+            Icon(
+                imageVector = Icons.Filled.Refresh,
+                tint = Color.White,
+                contentDescription = stringResource(id = R.string.flip_camera_content_description),
+                modifier = Modifier.size(72.dp)
+            )
+        }
+    }
+}
+
+@Composable
+fun SettingsNavButton(modifier: Modifier, onNavigateToSettings: () -> Unit) {
+    IconButton(
+        modifier = modifier,
+        onClick = onNavigateToSettings,
+    ) {
+        Icon(
+            imageVector = Icons.Filled.Settings,
+            tint = Color.White,
+            contentDescription = stringResource(R.string.settings_content_description),
+            modifier = Modifier.size(72.dp)
+        )
+    }
+}
+
+@Composable
+fun ZoomScaleText(zoomScale: Float, show: Boolean) {
+    val contentAlpha = animateFloatAsState(
+        targetValue = if (show) 1f else 0f, label = "zoomScaleAlphaAnimation",
+        animationSpec = tween()
+    )
+    Text(
+        modifier = Modifier.alpha(contentAlpha.value),
+        text = String.format("%.1fx", zoomScale),
+        fontSize = 20.sp,
+        color = Color.White
+    )
+}
+
+@Composable
+fun CaptureButton(
+    modifier: Modifier = Modifier,
```

Comment:
nit: when we pass in modifier to a composable, it should be the first optional parameter, but comes after non-optional parameters.

In this case, we should think about if we need to pass in a modifier for this, since we're not planning on re-using this element with modifier changes that affect the element internally.

Reply:
that's true in this case. we don't need the modifier here.

however if we may want to change the attributes of this (or any) composable's placement with respect to its parent container, we will need to designate those modifier values and pass it in.

 It also makes it easier to change the spacing/padding attributes up here vs sifting through the components file every time.

Reply:
Sounds good, lets keep the modifiers.

We should make them the first optional parameter.